In [4]:
def hello(name):
    print('{} 님 안녕하세요'.format(name))


def niceMeet(name):
    print('{} 님 만나서 반갑습니다.'.format(name))

In [5]:
hello('김연아')

김연아 님 안녕하세요


In [6]:
niceMeet('김연아')

김연아 님 만나서 반갑습니다.


In [7]:
import greet

In [8]:
greet.hello('김연아')

김연아 님 안녕하세요


In [9]:
greet.niceMeet('김연아')

김연아 님 만나서 반갑습니다.


In [10]:
from greet import hello, niceMeet

In [11]:
hello('홍길동')

홍길동 님 안녕하세요


In [12]:
niceMeet('홍길동')

홍길동 님 만나서 반갑습니다.


In [13]:
import math

In [19]:
a = 3.6
print('math.floor(a) :' + str(math.floor(a)))
print('math.ceil(a) :' + str(math.ceil(a)))
print('round(a) :' + str(round(a)))
print('math.factorial(round(a) :' + str(math.factorial(round(a))))

math.floor(a) :3
math.ceil(a) :4
round(a) :4
math.factorial(round(a) :24


In [25]:
b = math.sin(math.pi/4)
print(b)
b = math.cos(math.pi/2)
print(b)
b = math.tan(math.pi/6)
print(b)

0.7071067811865476
6.123233995736766e-17
0.5773502691896257


In [29]:
c = math.pow(2,3)
print(c)
c = math.sqrt(144)
print(c)
c = math.log10(1000)
print(c)

8.0
12.0
3.0


In [59]:
import random

In [93]:
x = random.randint(1,6)
print(x)
y = random.randint(1,6)
print(y)

6
1


In [6]:
import random
msg = {1: '가위', 2: '바위', 3: '보'}
while True:
    me = int(input('가위-1, 바위-2, 보-3  (0-종료)  : '))
    if me == 0:
        break
    ai = random.randint(1, 3)
    print('인간 vs AI 결과')
    print(' {}      {}'.format(msg[me], msg[ai]))
    if me > ai:
        print('당신의 승리입니다.')
    elif me < ai:
        print('AI의 승리입니다.')
    else:
        print('무승부입니다.')

가위-1, 바위-2, 보-3  (0-종료)  : 1
인간 vs AI 결과
 가위      보
AI의 승리입니다.
가위-1, 바위-2, 보-3  (0-종료)  : 0


# datetime module

In [1]:
from datetime import datetime
import time

In [2]:
today = datetime.now()
print(today)
print(type(today))

2023-03-16 13:21:36.747028
<class 'datetime.datetime'>


In [8]:
from datetime import datetime
import time

pre = datetime.now()
print(pre)
print('year is {}'.format(pre.year))
print('month is {}'.format(pre.month))
print('day is {}'.format(pre.day))
print('hour is {}'.format(pre.hour))
print('minute is {}'.format(pre.minute))
print('second is {}'.format(pre.second))
print('microsecond is {:06d}'.format(pre.microsecond))

time.sleep(5)

now = datetime.now()
print(now)
print('year is {}'.format(now.year))
print('month is {}'.format(now.month))
print('day is {}'.format(now.day))
print('hour is {}'.format(now.hour))
print('minute is {}'.format(now.minute))
print('second is {}'.format(now.second))
print('microsecond is {:06d}'.format(now.microsecond))

print('now - pre time is : {}'.format(now-pre))

2023-03-16 13:25:54.606938
year is 2023
month is 3
day is 16
hour is 13
minute is 25
second is 54
microsecond is 606938
2023-03-16 13:25:59.610190
year is 2023
month is 3
day is 16
hour is 13
minute is 25
second is 59
microsecond is 610190
now - pre time is : 0:00:05.003252


In [10]:
from datetime import datetime
import time
today = datetime.now()
print(today)
today_str = today.strftime('%Y/%m/%d%H:%M:%S')

2023-03-16 13:28:27.220282


# Logger Fuction 작성해 보세요
형식.... logging.csv
시간                           사람          AI      승리
2023-03-16 13:28:27.220282     가위          보      사람
2023-03-16 13:28:28.220282     바위          바위    무승부
2023-03-16 13:28:29.220282     보            가위    AI
2023-03-16 13:28:30.220282     가위          보      사람
//csv
open    logging.csv
write   시간,사람,AI,승리\n
close
//재료
file i/o
function (사람, AI)
   return 승리자
//로직
while 
   input 사람
   radom AI
   call function()
   save log

# 코딩...? 제일 먼저 무엇을 시작해야 하나요?

In [4]:
from datetime import datetime
import random

# write Header
file = open('log.csv', 'w')
file.write('시간,사람,AI,승리\n')
file.close()


human = '가위'
ai = '보'
winner = '사람'


def chatKMJ(human, ai):
    if(ai == '가위'):
        if(human == '바위'):
            result = '사람'
        elif(human == '가위'):
            result = '무승부'
        elif(human == '보'):
            result = 'AI'
    elif(ai == '보'):
        if(human == '가위'):
            result = '사람'
        elif(human == '보'):
            result = '무승부'
        elif(human == '바위'):
            result = 'AI'
    elif(ai == '바위'):
        if(human == '보'):
            result = '사람'
        elif(human == '바위'):
            result = '무승부'
        elif(human == '가위'):
            result = 'AI'
    return result


# write log
while True:
    human = input('가위,바위,보')
    ai = random.choice(['가위', '바위', '보'])

    if(human == '종료'):
        print('종료합니다')
        break
    elif not(human == '가위' or human == '바위' or human == '보'):
        print('continue')
        continue

    winner = chatKMJ(human, ai)

    today = datetime.now()
    file = open('log.csv', 'a')
    file.write('\'{},{},{},{}\n'.format(today, human, ai, winner))
    file.close()

가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보ㅍ
continue
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보ㅍ
continue
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바위,보가위
가위,바

In [5]:
print('Thank you chatKMJ')

Thank you chatKMJ


# Thank you chatKMJ

In [6]:
chatKMJ > ... 덧셈 교환 법칙 증명 코드를 작성해줘

SyntaxError: invalid syntax (218214083.py, line 1)

In [8]:
a = int(input('1st 입력 : '))
b = int(input('2nd 입력 : '))
c = a+b
d = b+a
if c == d:
    print('{}+{}={}'.format(a, b, c))
    print('{}+{}={}'.format(b, a, d))
    print('덧셈의 교환법칙이 성립합니다.')
else:
    print('덧셈의 교환법칙이 성립하지 않습니다.')

1st 입력 : 10
2nd 입력 : 20
10+20=30
20+10=30
덧셈의 교환법칙이 성립합니다.


# 바코드 암호화 / 복호화 코드를 작성해줘

In [ ]:
123 -> :::||::|:|::||: 으로 출력
:::||::|:|::||: -> 123 으로 출력 하는 코드를 작성해주세요

In [ ]:
123 -> :::||    ::|:|      ::||:

In [50]:
secretCode = ['||:::', ':::||', '::|:|', '::||:', ':|::|',
              ':|:|:', ':||::', '|:::|', '|::|:', '|:|::']
print(secretCode[0], secretCode[1])
print(type(secretCode))
'''secretCode = {0:'||:::',1:':::||',2:'::|:|',3:'::||:',4:':|::|',
              5:':|:|:',6:':||::',7:'|:::|',8:'|::|:',9:'|:|::'}
print(secretCode[0],secretCode[1])
print(type(secretCode))'''

||::: :::||
<class 'list'>


"secretCode = {0:'||:::',1:':::||',2:'::|:|',3:'::||:',4:':|::|',\n              5:':|:|:',6:':||::',7:'|:::|',8:'|::|:',9:'|:|::'}\nprint(secretCode[0],secretCode[1])\nprint(type(secretCode))"

# 1단계 : 123 -> :::||::|:|::||:

In [66]:
strNums = '123'

# print(type(strNums))#int -> string ... char char char... char
# 1
#print('[1] '+secretCode[1])
# 2
#print('[2] '+secretCode[2])
# 3
#print('[3] '+secretCode[3])


def ConvertNumsToBarCode(strNums):
    strBarCodes = ''
    for n in strNums:
        # print(n)
        # print(type(n))
        #print('{} '.format(n)+secretCode[int(n)])
        strBarCodes = strBarCodes + secretCode[int(n)]
    #print(strNums + '->'+ strBarCodes)
    return strBarCodes


strBarCodes = ConvertNumsToBarCode(strNums)
print(strNums + '->' + strBarCodes)
# 1단계 : 123 -> :::||::|:|::||:

123->:::||::|:|::||:


In [68]:
strNums = '123'
def ConvertNumsToBarCode(strNums):
    strBarCodes = ''
    for n in strNums:
        strBarCodes += secretCode[int(n)]
    return strBarCodes

strBarCodes = ConvertNumsToBarCode(strNums)
print(strNums + '->'+ strBarCodes)

123->:::||::|:|::||:


# 2단계 : :::||::|:|::||: -> 123

In [95]:
strBarcodes = ':::||::|:|::||:'

print('[0] :'+secretCode[0] + ',   [0] Size : ' + str(len(secretCode[0])))
print('='*70)


def ConvertBarCodeToNums(strBarCodes):
    print('function param : ' + strBarCodes)
    #:::||::|:|::||: ... input 15 ... inputSize
    #:::||           ... div    5 ... ptrnSize
    #                    step   3 ... stepSize
    intputSize = len(strBarCodes)  # 15
    ptrnSize = len(secretCode[0])  # 5
    stepSize = int(intputSize / ptrnSize)  # 3
    print('intputSize - ' + str(intputSize))
    print('ptrnSize - ' + str(ptrnSize))
    print('stepSize - ' + str(stepSize))

    # for(int n=0; n<inputSize; n+=stepSize)
    for n in range(0, intputSize, ptrnSize):
        # ptrn1 = strBarCodes[0:0+5]#0->n, 5->ptrnSize
        ptrn = strBarCodes[n:n+ptrnSize]  # 0->n, 5->ptrnSize
        print('sub string ptrn - ' + ptrn)

    # var -> for iterable
    # ptrn1 = strBarCodes[0:0+5]#1   jump 5
    # ptrn2 = strBarCodes[5:5+5]#2   jump 5
    # ptrn3 = strBarCodes[10:10+5]#3

    #print('sub string ptrn1 - ' + ptrn1)


ConvertBarCodeToNums(strBarCodes)

[0] :||:::,   [0] Size : 5
function param : :::||::|:|::||:
intputSize - 15
ptrnSize - 5
stepSize - 3
sub string ptrn - :::||
sub string ptrn - ::|:|
sub string ptrn - ::||:


In [113]:
strBarcodes = ':::||::|:|::||:'


def ConvertBarCodeToNums1(strBarCodes):
    intputSize = len(strBarCodes)  # 15
    ptrnSize = len(secretCode[0])  # 5
    strfindNums = ''
    for n in range(0, intputSize, ptrnSize):
        ptrn = strBarCodes[n:n+ptrnSize]  # 0->n, 5->ptrnSize
        idx = 0
        find_idx = 0
        # get index -> value ... 0 index....0 value [LIST]
        for code in secretCode:
            if code == ptrn:
                find_idx = idx
                strfindNums += str(find_idx)
                break
            else:
                idx += 1
    return strfindNums


def ConvertBarCodeToNums2(strBarcodes):
    ptrnSz = len(secretCode[0])
    ptrnCount = len(strBarcodes)/ptrnSz
    ptrnNums = ''
    for x in range(0, len(strBarcodes), ptrnSz):
        ptrn = strBarcodes[x:x+ptrnSz]
        for k in range(len(secretCode)):
            if ptrn == secretCode[k]:
                ptrnNums += str(k)
    return ptrnNums


print('Barcodes - {}'.format(strBarcodes))
strNums = ConvertBarCodeToNums2(strBarCodes)
print('Nums - {}'.format(strNums))

Barcodes - :::||::|:|::||:
Nums - 123


In [ ]:
#input output 출력까지 완료하세요

In [1]:
'''[index access] for(int i=0
                   i < 10
                   i++) = > for i in range(0, 10, 1)...iterable
for i in range(len(secretCodes))


[pointer] for(auto i=v.begin()
              i != v.end()
              i++)
std: : vector < int > v
v[0]
v.begin()
for x in secretCodes:  # secretCodes   ...iterable'''

SyntaxError: invalid syntax (288013870.py, line 1)

In [3]:
secret = ['||:::', ':::||', '::|:|', '::||:', ':|::|',
          ':|:|:', ':||::', '|:::|', '|::|:', '|:|::']
print(secret)

['||:::', ':::||', '::|:|', '::||:', ':|::|', ':|:|:', ':||::', '|:::|', '|::|:', '|:|::']


In [4]:
strNums = '528'


def ConvertNumsToBarcodes(strNums):
    strBarcodes = ''
    '''for i,n in enumerate(strNums):#i-index, n-value via Tuple
        print(i,n)
        strBarcodes+=barCodes[i]'''
    '''for i in range(len(strNums)):
        strBarcodes+=barCodes[i]'''
    for i in strNums:
        print(i)  # i - str , in strNums
        i = int(i)  # convert str to int
        strBarcodes += secret[i]
    return strBarcodes


print(ConvertNumsToBarcodes(strNums))

5
2
8
:|:|:::|:||::|:


In [5]:
strBarcodes = ':|:|:::|:||::|:'


def ConvertBarcodesToNums(strBarcodes):
    ptrnSz = len(secret[0])
    ptrnCount = len(strBarcodes)/ptrnSz
    ptrnNums = ''
    for x in range(0, len(strBarcodes), ptrnSz):
        print(x, x+ptrnSz)
        ptrn = strBarcodes[x:x+ptrnSz]
        print(ptrn)
        for k in range(len(secret)):
            if ptrn == secret[k]:
                ptrnNums += str(k)
    return ptrnNums


print('result : ' + str(ConvertBarcodesToNums(strBarcodes)))

0 5
:|:|:
5 10
::|:|
10 15
|::|:
result : 528
